src/url.js

```diff
@@ -650,3 +650,17 @@ export function checkCorsUrl(url) {
 export function tryDecodeUriComponent(component, opt_fallback) {
   return tryDecodeUriComponent_(component, opt_fallback);
 }
+
+/**
+ * Adds the path to the given url.
+ *
+ * @param {!HTMLAnchorElement} a
+ * @param {string} url
+ * @param {string} path
+ * @return {string}
+ */
+export function appendPathToUrlWithA(a, url, path) {
```

Comment:
Why can't we use the URL service here instead of creating an anchor and passing it in? The point of the URL service is to avoid creating unnecessary `<a>` elements and have caching of URLs.

Reply:
Done!